packages/gatsby-plugin-sharp/src/gatsby-node.js

```diff
@@ -1,10 +1,84 @@
-const { setBoundActionCreators, setPluginOptions } = require(`./index`)
+const {
+  setBoundActionCreators,
+  setPluginOptions,
+  queue,
+  reportError,
+} = require(`./index`)
+const processFile = require(`./processFile`)
+const { scheduleJob } = require(`./scheduler`)
 
-exports.onPreInit = ({ actions }, pluginOptions) => {
+const getQueueFromCache = async cache => {
+  const rawQueue = await cache.get(`queue`)
+
+  if (!rawQueue) {
+    return new Map()
+  }
+
+  return new Map(rawQueue)
+}
+
+exports.onPreInit = async ({ actions }, pluginOptions) => {
   setBoundActionCreators(actions)
   setPluginOptions(pluginOptions)
 }
 
+/**
+ * save queue to the cache
+ */
+exports.onPostBootstrap = async ({ cache }) => {
```

Comment:
in `develop` we can run queries after bootstrap (especially when working with files and editing them while `develop` is running) - it's good idea to wait for this event for initial save - but after that we would need to re-save queue

Reply:
Any suggestions? I Probably need to save it when dev server is created, just before [line 99](https://github.com/gatsbyjs/gatsby/pull/10964/files#diff-f60f2b266f94ca7cde8437d8451cfe39R66).

If queries change is there any mechanism to know what nodes are being removed because I probably want to remove unused images.